genrl/bandit/agents/cb_agents/bootstrap_neural.py

```diff
@@ -0,0 +1,120 @@
+from typing import Optional
+
+import numpy as np
+import torch
+
+from genrl.bandit.agents.cb_agents.base import DCBAgent
+from genrl.bandit.agents.cb_agents.common import NeuralBanditModel, TransitionDB
+from genrl.bandit.bandits.data_bandits.base import DataBasedBandit
+
+
+class BootstrapNeuralAgent(DCBAgent):
+    """Bootstraped ensemble agentfor deep contextual bandits.
+
+    Args:
+        bandit (DataBasedBandit): The bandit to solve
+        init_pulls (int, optional): Number of times to select each action initially.
+            Defaults to 3.
+        hidden_dims (List[int], optional): Dimensions of hidden layers of network.
+            Defaults to [50, 50].
+        init_lr (float, optional): Initial learning rate. Defaults to 0.1.
+        lr_decay (float, optional): Decay rate for learning rate. Defaults to 0.5.
+        lr_reset (bool, optional): Whether to reset learning rate ever train interval.
+            Defaults to True.
+        max_grad_norm (float, optional): Maximum norm of gradients for gradient clipping.
+            Defaults to 0.5.
+        dropout_p (Optional[float], optional): Probability for dropout. Defaults to None
+            which implies dropout is not to be used.
+        eval_with_dropout (bool, optional): Whether or not to use dropout at inference.
+            Defaults to False.
+        n (int, optional): Number of models in ensemble. Defaults to 10.
+        add_prob (float, optional): Probability of adding a transition to a database.
+            Defaults to 0.95.
+        device (str): Device to use for tensor operations.
+            "cpu" for cpu or "cuda" for cuda. Defaults to "cpu".
+    """
+
+    def __init__(self, bandit: DataBasedBandit, **kwargs):
+        super(BootstrapNeuralAgent, self).__init__(bandit, kwargs.get("device", "cpu"))
+        self.init_pulls = kwargs.get("init_pulls", 3)
+        self.n = kwargs.get("n", 10)
+        self.add_prob = kwargs.get("add_prob", 0.95)
+        self.models = [
+            NeuralBanditModel(
+                context_dim=self.context_dim,
+                hidden_dims=kwargs.get("hidden_dims", [50, 50]),
+                n_actions=self.n_actions,
+                init_lr=kwargs.get("init_lr", 0.1),
+                max_grad_norm=kwargs.get("max_grad_norm", 0.5),
+                lr_decay=kwargs.get("lr_decay", 0.5),
+                lr_reset=kwargs.get("lr_reset", True),
+                dropout_p=kwargs.get("dropout_p", None),
+            )
```

Comment:
Not necessary? You could just send **kwargs instead.

Reply:
THought it would be better to be more explicit here

Reply:
Then it would be better not to use kwargs.get() 

Pass the proper arguments instead

Reply:
That would just unncessarily add more lines to store the vars seperately right?